src/main/java/org/opensearch/security/hasher/BCryptPasswordHasher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.hasher;
+
+import java.nio.CharBuffer;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Arrays;
+
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.SpecialPermission;
+
+import com.password4j.BcryptFunction;
+import com.password4j.HashingFunction;
+import com.password4j.Password;
+import com.password4j.types.Bcrypt;
+
+public class BCryptPasswordHasher implements PasswordHasher {
+
+    @Override
+    public String hash(char[] password) {
+        if (password == null || password.length == 0) {
```

Comment:
Lest use OS API for checking the `password` variable:
`org.opensearch.core.common.Strings.isNullOrEmpty`

Reply:
That would require converting to a String first which I am trying to avoid. However will use it for checking the hash variable

Reply:
Ahhh sorry, it uses `char[]` did not notice.